libkmod/libkmod-index.c

```diff
@@ -670,86 +666,59 @@ static inline const char *read_chars_mm(const void **p, size_t *rlen)
 	return addr;
 }
 
-static struct index_mm_node *index_mm_read_node(struct index_mm *idx, uint32_t offset)
+static inline void read_value_mm(const void **p, struct index_mm_value *v)
+{
+	v->priority = read_u32_mm(p);
+	v->value = read_chars_mm(p, &v->len);
+}
+
+/* reads node into given node struct and returns its address on success or NULL on error. */
+static struct index_mm_node *index_mm_read_node(struct index_mm *idx, uint32_t offset,
+						struct index_mm_node *node)
```

Comment:
My first version used `bool` and bloated the machine code. Both, clang and gcc, are rather bad in handling bool return values, because they start swapping registers. They are quite optimized in handling pointers directly instead of reloading them.

At least it's in sync with `fgets`, although I dislike that function.